src/app/tests/TestCommissioningWindowManager.cpp

```diff
@@ -406,4 +409,128 @@ TEST_F(TestCommissioningWindowManager, TestCheckCommissioningWindowManagerEnhanc
+
+    commissionMgr.OnPlatformEvent(&event);
+    EXPECT_TRUE(commissionMgr.IsCommissioningWindowOpen());
+    EXPECT_EQ(commissionMgr.GetCommissioningMode(), chip::Dnssd::CommissioningMode::kEnabledBasic);
+    commissionMgr.CloseCommissioningWindow();
+    EXPECT_EQ(commissionMgr.GetCommissioningMode(), chip::Dnssd::CommissioningMode::kDisabled);
+}
+
+// Verify that PASE session is properly evicted when failsafe timer is expired
+TEST_F(TestCommissioningWindowManager, TestOnPlatformEventFailSafeTimerExpiredPASEEVicted)
+{
+    CommissioningWindowManager & commissionMgr = Server::GetInstance().GetCommissioningWindowManager();
+    auto access                                = chip::Test::CommissioningWindowManagerTestAccess(&commissionMgr);
+    auto & sessionMgr                          = Server::GetInstance().GetSecureSessionManager();
+
+    auto & paseSession            = access.GetPASESession();
+    uint16_t localSessionId       = 1;
+    chip::NodeId peerNodeId       = chip::kUndefinedNodeId;
+    uint16_t peerSessionId        = 2;
+    chip::FabricIndex fabricIndex = chip::kUndefinedFabricIndex;
+    chip::Transport::PeerAddress peerAddress =
+        chip::Transport::PeerAddress::UDP(chip::Inet::IPAddress::Loopback(chip::Inet::IPAddressType::kAny), CHIP_PORT);
+
+    auto role = chip::CryptoContext::SessionRole::kResponder;
+
+    EXPECT_EQ(commissionMgr.OpenBasicCommissioningWindow(commissionMgr.MaxCommissioningTimeout(),
+                                                         CommissioningWindowAdvertisement::kDnssdOnly),
+              CHIP_NO_ERROR);
+
+    // Inject a fake PASE session into SessionManager
+    CHIP_ERROR err = sessionMgr.InjectPaseSessionWithTestKey(paseSession, localSessionId, peerNodeId, peerSessionId, fabricIndex,
+                                                             peerAddress, role);
+    EXPECT_EQ(err, CHIP_NO_ERROR);
+    EXPECT_TRUE(paseSession.Get().HasValue());
+    EXPECT_TRUE(paseSession->AsSecureSession()->IsActiveSession());
+
+    auto event = CreateEvent(chip::DeviceLayer::DeviceEventType::kFailSafeTimerExpired);
+
+    commissionMgr.OnPlatformEvent(&event);
+    // Verify that PASE Session was evicted after failsafe timer expiration
+    EXPECT_FALSE(paseSession.Get().HasValue());
+    commissionMgr.CloseCommissioningWindow();
+    EXPECT_EQ(commissionMgr.GetCommissioningMode(), chip::Dnssd::CommissioningMode::kDisabled);
+}
+
+// Verify that operational advertising is started when the operational network is enabled
+TEST_F(TestCommissioningWindowManager, TestOnPlatformEventOperationalNetworkEnabled)
+{
+    CommissioningWindowManager & commissionMgr = Server::GetInstance().GetCommissioningWindowManager();
+
+    auto event = CreateEvent(chip::DeviceLayer::DeviceEventType::kOperationalNetworkEnabled);
+
+    commissionMgr.OnPlatformEvent(&event);
+    EXPECT_EQ(chip::app::DnssdServer::Instance().AdvertiseOperational(), CHIP_NO_ERROR);
```

Comment:
This isn't checking that the commissioning manager started the advertisement - it's starting the advertisement.